plugin/storage/es/factory.go

```diff
@@ -241,8 +241,15 @@ func createSpanWriter(
 	if err != nil {
 		return nil, err
 	}
+
+	elasticsearch8Client, err := config.NewElasticSearch8Client(cfg, logger)
```

Comment:
Should this be done conditionally on the desired version? Would this not fail if the actual backend is <= v7?

Reply:
I am not sure why it would fail, because in the local testing , the templates do get added. If the actual backend is <=v7 there would be a client declaration and we would just not use it anywhere. We can do it conditionally but I dont see where it would fail if the backend in <=v7.

Reply:
I suppose it will come up in integration tests if it fails